packages/filesystem/src/browser/file-tree/file-tree-widget.tsx

```diff
@@ -152,14 +162,20 @@ export class FileTreeWidget extends TreeWidget {
             }
         }
     }
-
-    protected setTreeNodeAsData(data: DataTransfer, node: TreeNode): void {
-        data.setData('tree-node', node.id);
+    protected setTreeNodeAsData(data: DataTransfer, sourceNode: TreeNode, relatedNodes: TreeNode[]): void {
+        data.setData('tree-node', sourceNode.id);
+        data.setData('selected-tree-nodes', JSON.stringify(relatedNodes.map(node => node.id)));
     }
-
-    protected getTreeNodeFromData(data: DataTransfer): TreeNode | undefined {
```

Comment:
please always declare return types in order to avoid surprising breaking changes by changing method body